compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -32,28 +32,38 @@ use super::brillig_fn::FunctionContext;
 use super::constant_allocation::InstructionLocation;
 
 /// Generate the compilation artifacts for compiling a function into brillig bytecode.
-pub(crate) struct BrilligBlock<'block> {
-    pub(crate) function_context: &'block mut FunctionContext,
+pub(crate) struct BrilligBlock<'block, 'global, Registers: RegisterAllocator> {
```

Comment:
Do we need two globals here? I'd expect that the globals would outlive `'block` anyway so we can just reuse that lifetime.

Reply:
Huh, not sure why I put them on the `FunctionContext`, good point. I think with how I have it now with the globals inside of `FunctionContext` I cannot avoid the second lifetime, but I will move it out of there and put it directly on `BrilligBlock`.

Reply:
Done, I now put the globals directly on the block.